ROS2: general macro for RGB-D cameras

TomasMerva opened the pull request:
Addressing https://github.com/ctu-mrs/mrs_uav_gazebo_simulator/issues/23

spurnvoj commented:
@TomasMerva not sure if you know about this discussion regarding RGBD sensor frames differences https://github.com/gazebosim/gz-sensors/issues/488 and https://github.com/gazebosim/ros_gz/pull/703

TomasMerva commented:
**This update includes:**
- Updated RGB and depth camera templates with optical frame support to align CameraInfo with sensor data
- Refactored `sdf_to_tf_publisher.py` to support optical frames for cameras

**RGB-D camera template (pending bug fix):**
- The issue is that even when specifying the additional optical frame, the point cloud and `CameraInfo` are still misaligned due to differences in frame conventions, as discussed in more detail https://github.com/gazebosim/gz-sensors/issues/488.
<p align="center">
<img width="752" height="400" alt="Screenshot from 2025-11-05 10-05-45" src="https://github.com/user-attachments/assets/bf4914f5-81ff-422f-b03b-5c339aed6650" />
</p>

- The template is prepared and we are successfully publishing point cloud data to ROS. However, users should be aware that the point cloud data must be manually transformed, which is not included in this PR.

spurnvoj commented on models/mrs_robots_description/sdf/README.md:
````diff
@@ -0,0 +1,48 @@
+{%- set camera_gz_frame_id = spawner_args['name'] + '/' + camera_name + '/color_optical' -%}
+```
+
+The resulting camera-related topics will look like as follows:
````
This is relevant to the gazebo topic; perhaps we could also discuss how it is implemented on the ROS side.

matemat13 commented:
Let me know when this is ready for me to check the correctness of the camera parameters. Also, I assume this won't get backported to ROS1? :(

spurnvoj commented:
Hi @matemat13, this pull request has been merged. You can now check the correctness of the camera parameters. Additionally, the Luxonix camera has been included in this pull request.
